python/ModelTools.py

```diff
@@ -91,6 +95,28 @@ def doModel(self):
             if self.options.verbose > 2: 
                 self.out.pdf("model_s").graphVizTree(self.options.out+".dot", "\\n")
                 print "Wrote GraphVizTree of model_s to ",self.options.out+".dot"
+
+    def doRateParams(self):
+	# First do independant parameters, then expressions
+	for rp in self.DC.rateParams.keys():
+	  type = self.DC.rateParams[rp][-1]
+	  if type!=0: continue
+	  argu,argv = self.DC.rateParams[rp][0],self.DC.rateParams[rp][1]
+	  if self.out.arg(argu): continue
+
+	  self.doVar("%s[%s,0,1]"%(argu,argv))
+	  self.out.var(argu).removeRange()
```

Comment:
Currently the syntax does not support ranges, these of course can be set in
the command line as with any other parameters.


On Tue, Jul 28, 2015 at 10:22 AM, André David <notifications@github.com>
wrote:

> In python/ModelTools.py
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/230#discussion_r35624219>
> :
>
> > @@ -91,6 +95,28 @@ def doModel(self):
> >              if self.options.verbose > 2:
> >                  self.out.pdf("model_s").graphVizTree(self.options.out+".dot", "\\n")
> >                  print "Wrote GraphVizTree of model_s to ",self.options.out+".dot"
> > +
> > +    def doRateParams(self):
> > +	# First do independant parameters, then expressions
> > +	for rp in self.DC.rateParams.keys():
> > +	  type = self.DC.rateParams[rp][-1]
> > +	  if type!=0: continue
> > +	  argu,argv = self.DC.rateParams[rp][0],self.DC.rateParams[rp][1]
> > +	  if self.out.arg(argu): continue
> > +
> > +	  self.doVar("%s[%s,0,1]"%(argu,argv))
> > +	  self.out.var(argu).removeRange()
>
> Should the syntax foresee ranges, or would it be dangerous to hit
> boundaries?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/230/files#r35624219>
> .
>